File: app/src/main/java/com/apnagodam/staff/activity/in/truckbook/TruckBookListingActivity.java
package com.apnagodam.staff.activity.in.truckbook;

import android.graphics.Rect;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.Window;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.appcompat.app.AlertDialog;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.apnagodam.staff.Base.BaseActivity;
import com.apnagodam.staff.Network.NetworkCallback;
import com.apnagodam.staff.R;
import com.apnagodam.staff.activity.StaffDashBoardActivity;
import com.apnagodam.staff.adapter.TruckBookAdapter;
import com.apnagodam.staff.databinding.ActivityListingBinding;
import com.apnagodam.staff.module.AllTruckBookListResponse;
import com.apnagodam.staff.utils.PaginationScrollListener;
import java.util.List;


public class TruckBookListingActivity extends BaseActivity<ActivityListingBinding> {
    private List<AllTruckBookListResponse.TruckBookCollection> AllCases;
    private TruckBookAdapter truckBookAdapter;

    private int pageOffset = 1;
    private int totalPage = 0;
    private Boolean isLastPage = false;
    private Boolean isLoading = false;

    @Override
    protected int getLayoutResId() {
        return R.layout.activity_listing;
    }

    @Override
    protected void setUp() {
        setSupportActionBar(binding.toolbar);
        binding.titleHeader.setText(getResources().getString(R.string.truck_book));
        binding.tvId.setText(getResources().getString(R.string.case_idd));
        binding.tvMoreView.setText(getResources().getString(R.string.more_view_truck));
        binding.tvPhone.setText(getResources().getString(R.string.truck_book));
        getSupportActionBar().setDisplayShowTitleEnabled(false);

        setAdapter();

        getAllCases();
        binding.ivClose.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivityAndClear(StaffDashBoardActivity.class);
            }
        });
    }

    private void setAdapter() {
        binding.rvDefaultersStatus.addItemDecoration(new DividerItemDecoration(TruckBookListingActivity.this, LinearLayoutManager.VERTICAL));
        binding.rvDefaultersStatus.setHasFixedSize(true);
        LinearLayoutManager horizontalLayoutManager = new LinearLayoutManager(TruckBookListingActivity.this, LinearLayoutManager.VERTICAL, false);
        binding.rvDefaultersStatus.setLayoutManager(horizontalLayoutManager);
        truckBookAdapter=new TruckBookAdapter(AllCases, TruckBookListingActivity.this);
        binding.rvDefaultersStatus.setAdapter(truckBookAdapter);
        pagination(horizontalLayoutManager);

    }

    private void pagination(LinearLayoutManager horizontalLayoutManager) {
        binding.rvDefaultersStatus.addOnScrollListener(new PaginationScrollListener(horizontalLayoutManager) {
            @Override
            public boolean isLastPage() {
                return isLastPage;
            }

            @Override
            public boolean isLoading() {
                return isLoading;
            }

            @Override
            public void loadMoreItems() {
                isLoading = true;
                getMoreItems();
            }
        });
    }

    private void getMoreItems() {
        if (isLoading && pageOffset<totalPage) {
            pageOffset = pageOffset + 1;
            binding.layoutLoader.setVisibility(View.VISIBLE);
            getAllCases();
            isLoading = false;
        }else{
            binding.layoutLoader.setVisibility(View.GONE);
        }
    }


    private void getAllCases() {
        apiService.getTruckBookList("15",pageOffset).enqueue(new NetworkCallback<AllTruckBookListResponse>(getActivity()) {
            @Override
            protected void onSuccess(AllTruckBookListResponse body) {
                if (body.getTruckBookCollection() == null || body.getTruckBookCollection().isEmpty()) {
                    binding.txtemptyMsg.setVisibility(View.VISIBLE);
                    binding.layoutLoader.setVisibility(View.GONE);
                    binding.rvDefaultersStatus.setVisibility(View.GONE);
                } else {
                    AllCases.addAll(body.getTruckBookCollection());
                    truckBookAdapter.notifyDataSetChanged();
                }
            }
        });
    }

    public void ViewData(int position) {
        Rect displayRectangle = new Rect();
        Window window = this.getWindow();
        window.getDecorView().getWindowVisibleDisplayFrame(displayRectangle);
        final AlertDialog.Builder builder = new AlertDialog.Builder(TruckBookListingActivity.this, R.style.CustomAlertDialog);
        LayoutInflater inflater = TruckBookListingActivity.this.getLayoutInflater();
        View dialogView = inflater.inflate(R.layout.lead_view, null);
        dialogView.setMinimumWidth((int) (displayRectangle.width() * 1f));
        dialogView.setMinimumHeight((int) (displayRectangle.height() * 1f));
        builder.setView(dialogView);
        final AlertDialog alertDialog = builder.create();
        ImageView cancel_btn = dialogView.findViewById(R.id.cancel_btn);
        TextView lead_id = dialogView.findViewById(R.id.lead_id);
        TextView genrated_by = dialogView.findViewById(R.id.genrated_by);
        TextView customer_name = dialogView.findViewById(R.id.customer_name);
        TextView phone_no = dialogView.findViewById(R.id.phone_no);
        TextView location_title = dialogView.findViewById(R.id.location_title);
        TextView commodity_name = dialogView.findViewById(R.id.commodity_name);
        TextView est_quantity_nmae = dialogView.findViewById(R.id.est_quantity_nmae);
        TextView terminal_name = dialogView.findViewById(R.id.terminal_name);
        TextView purpose_name = dialogView.findViewById(R.id.purpose_name);
        TextView commitemate_date = dialogView.findViewById(R.id.commitemate_date);
        TextView total_weights = dialogView.findViewById(R.id.vehicle_no);
        TextView total_bags = dialogView.findViewById(R.id.in_out);
        TextView create_date = dialogView.findViewById(R.id.create_date);
        TextView case_id = dialogView.findViewById(R.id.a1);
        TextView username = dialogView.findViewById(R.id.a2);
        TextView vehicle_no = dialogView.findViewById(R.id.a4);
        TextView processing_fees = dialogView.findViewById(R.id.a5);
        TextView inteerset_rate = dialogView.findViewById(R.id.a6);
        TextView transport_rate = dialogView.findViewById(R.id.a7);
        TextView loan = dialogView.findViewById(R.id.a8);
        TextView price = dialogView.findViewById(R.id.a9);
        TextView rent = dialogView.findViewById(R.id.a10);
        TextView labour_rent = dialogView.findViewById(R.id.a11);
        TextView total_weight = dialogView.findViewById(R.id.a12);
        TextView notes = dialogView.findViewById(R.id.a13);
        TextView bags = dialogView.findViewById(R.id.a14);
        LinearLayout price_extra = dialogView.findViewById(R.id.price_extra);
        price_extra.setVisibility(View.VISIBLE);
        LinearLayout case_extra = dialogView.findViewById(R.id.case_extra);
        case_extra.setVisibility(View.VISIBLE);
        LinearLayout truck_extra = dialogView.findViewById(R.id.truck_extra);
        truck_extra.setVisibility(View.VISIBLE);
        TextView converted_by = dialogView.findViewById(R.id.converted_by);
        TextView gate_pass = dialogView.findViewById(R.id.gate_pass);
        TextView user = dialogView.findViewById(R.id.user);
        TextView coldwin = dialogView.findViewById(R.id.coldwin);
        TextView purchase_details = dialogView.findViewById(R.id.purchase_details);
        TextView loan_details = dialogView.findViewById(R.id.loan_details);
        TextView selas_details = dialogView.findViewById(R.id.selas_details);
        TextView gate_passs = dialogView.findViewById(R.id.gate_passs);
        TextView total_trans_cost = dialogView.findViewById(R.id.total_trans_cost);
        TextView advance_patyment = dialogView.findViewById(R.id.advance_patyment);
        TextView start_date_time = dialogView.findViewById(R.id.start_date_time);
        TextView end_date_time = dialogView.findViewById(R.id.end_date_time);
        TextView settleememt_amount = dialogView.findViewById(R.id.settleememt_amount);
        gate_passs.setText("" + ((AllCases.get(position).getGatePass()) != null ? AllCases.get(position).getGatePass() : "N/A"));
        total_trans_cost.setText("" + ((AllCases.get(position).getTotalTransportCost()) != null ? AllCases.get(position).getTotalTransportCost() : "N/A"));
        advance_patyment.setText("" + ((AllCases.get(position).getAdvancePayment()) != null ? AllCases.get(position).getAdvancePayment() : "N/A"));
        start_date_time.setText("" + ((AllCases.get(position).getStartDateTime()) != null ? AllCases.get(position).getStartDateTime() : "N/A"));
        end_date_time.setText("" + ((AllCases.get(position).getEndDateTime()) != null ? AllCases.get(position).getEndDateTime() : "N/A"));
        settleememt_amount.setText("" + ((AllCases.get(position).getFinalSettlementAmount()) != null ? AllCases.get(position).getFinalSettlementAmount() : "N/A"));
        total_bags.setText("" + ((AllCases.get(position).getNoOfBags()) != null ? AllCases.get(position).getNoOfBags() : "N/A"));
        total_weights.setText("" + ((AllCases.get(position).getTotalWeight()) != null ? AllCases.get(position).getTotalWeight() : "N/A"));
        converted_by.setText("" + ((AllCases.get(position).getNotes()) != null ? AllCases.get(position).getNotes() : "N/A"));
        gate_pass.setText("Gaatepass/CDF Name : " + ((AllCases.get(position).getGatePassCdfUserName()) != null ? AllCases.get(position).getGatePassCdfUserName() : "N/A"));
        coldwin.setText("ColdWin Name: " + ((AllCases.get(position).getColdwinName()) != null ? AllCases.get(position).getColdwinName() : "N/A"));
        user.setText("User : " + ((AllCases.get(position).getFpoUserId()) != null ? AllCases.get(position).getFpoUserId() : "N/A"));
        purchase_details.setText("purchase Details : " + ((AllCases.get(position).getPurchaseName()) != null ? AllCases.get(position).getPurchaseName() : "N/A"));
        loan_details.setText("Loan Details : " + ((AllCases.get(position).getLoanName()) != null ? AllCases.get(position).getLoanName() : "N/A"));
        selas_details.setText("Sale Details : " + ((AllCases.get(position).getSaleName()) != null ? AllCases.get(position).getSaleName() : "N/A"));
        total_weight.setText(getResources().getString(R.string.total_weight));
        bags.setText(getResources().getString(R.string.bags));
        case_id.setText(getResources().getString(R.string.case_idd));
        notes.setText(getResources().getString(R.string.notes));
        rent.setText(getResources().getString(R.string.turnaround_time));
        labour_rent.setText(getResources().getString(R.string.commodity));
        loan.setText(getResources().getString(R.string.min_weight));
        transport_rate.setText(getResources().getString(R.string.transport_rate));
        price.setText(getResources().getString(R.string.max_weight));
        vehicle_no.setText(getResources().getString(R.string.vehicle_noa));
        processing_fees.setText(getResources().getString(R.string.driver_name));
        inteerset_rate.setText(getResources().getString(R.string.driver_phone_no));
        username.setText(getResources().getString(R.string.transporter));
        lead_id.setText("" + AllCases.get(position).getCaseId());
        genrated_by.setText("" + ((AllCases.get(position).getTransporter()) != null ? AllCases.get(position).getTransporter() : "N/A"));
        customer_name.setText("" + AllCases.get(position).getCustFname());
        location_title.setText("" + ((AllCases.get(position).getDriverName()) != null ? AllCases.get(position).getDriverName() : "N/A"));
        phone_no.setText("" + ((AllCases.get(position).getVehicleNo()) != null ? AllCases.get(position).getVehicleNo() : "N/A"));
       commodity_name.setText("" + ((AllCases.get(position).getDriverPhone()) != null ? AllCases.get(position).getDriverPhone() : "N/A"));
        est_quantity_nmae.setText("" + ((AllCases.get(position).getRatePerKm()) != null ? AllCases.get(position).getRatePerKm() : "N/A"));
        terminal_name.setText("" + ((AllCases.get(position).getMinWeight()) != null ? AllCases.get(position).getMinWeight() : "N/A"));
        purpose_name.setText("" + ((AllCases.get(position).getMaxWeight()) != null ? AllCases.get(position).getMaxWeight() : "N/A"));
        commitemate_date.setText("" + ((AllCases.get(position).getTurnaroundTime()) != null ? AllCases.get(position).getTurnaroundTime() : "N/A"));
        create_date.setText("" + ((AllCases.get(position).getCommodityId()) != null ? AllCases.get(position).getCommodityId() : "N/A"));
        cancel_btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                alertDialog.dismiss();
            }
        });
        alertDialog.show();
    }

    public void checkVeehicleNo(int postion) {
        Bundle bundle = new Bundle();
        bundle.putString("user_name", AllCases.get(postion).getCustFname());
        bundle.putString("case_id", AllCases.get(postion).getCaseId());
        bundle.putString("vehicle_no", AllCases.get(postion).getVehicleNo());
        startActivity(TruckUploadDetailsClass.class,bundle);
       /* apiService.cheeckvehiclePricicng(AllCases.get(postion).getCaseId()).enqueue(new NetworkCallback<VehcilePricingCheeck>(getActivity()) {
            @Override
            protected void onSuccess(VehcilePricingCheeck body) {
                if (body.getVichel_status().equalsIgnoreCase("1")) {
                    Toast.makeText(getApplicationContext(), body.getMessage(), Toast.LENGTH_LONG).show();
                } else {

                }
            }
        });*/
    }

}

File: app/src/main/java/com/apnagodam/staff/activity/in/truckbook/TruckBookListingActivity.kt
package com.apnagodam.staff.activity.`in`.truckbook

import android.app.Activity
import android.graphics.Rect
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AlertDialog
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout.OnRefreshListener
import com.apnagodam.staff.Base.BaseActivity
import com.apnagodam.staff.Network.NetworkCallbackWProgress
import com.apnagodam.staff.Network.NetworkResult
import com.apnagodam.staff.Network.viewmodel.TruckBookViewModel
import com.apnagodam.staff.R
import com.apnagodam.staff.activity.StaffDashBoardActivity
import com.apnagodam.staff.adapter.TruckBookAdapter
import com.apnagodam.staff.databinding.ActivityListingBinding
import com.apnagodam.staff.module.AllTruckBookListResponse
import com.apnagodam.staff.utils.Constants
import com.apnagodam.staff.utils.PhotoFullPopupWindow
import dagger.hilt.android.AndroidEntryPoint
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

@AndroidEntryPoint
class TruckBookListingActivity() : BaseActivity<ActivityListingBinding?>() {
    private var truckBookAdapter: TruckBookAdapter? = null
    private var pageOffset = 1
    private var totalPage = 0
    private var AllCases: MutableList<AllTruckBookListResponse.Datum?>? = null
    private var TruckImage: String? = null

    override fun getLayoutResId(): Int {
        return R.layout.activity_listing
    }
    private val truckViewModel  : TruckBookViewModel by viewModels<TruckBookViewModel>()

    override fun setUp() {
        AllCases = arrayListOf()
        getAllCases("")

        setAdapter()
        setSupportActionBar(binding!!.toolbar)
        binding!!.titleHeader.text = resources.getString(R.string.truck_book)
        binding!!.tvId.text = resources.getString(R.string.case_idd)
        binding!!.tvMoreView.text = resources.getString(R.string.more_view_truck)
        binding!!.tvPhone.text = resources.getString(R.string.truck_book)
        supportActionBar!!.setDisplayShowTitleEnabled(false)
        /* binding.rvDefaultersStatus.addItemDecoration(new DividerItemDecoration(TruckBookListingActivity.this, LinearLayoutManager.VERTICAL));
        LinearLayoutManager horizontalLayoutManager = new LinearLayoutManager(TruckBookListingActivity.this, LinearLayoutManager.VERTICAL, false);
        binding.rvDefaultersStatus.setLayoutManager(horizontalLayoutManager);*/
        binding!!.swipeRefresherStock.setOnRefreshListener(OnRefreshListener { getAllCases("") })
        binding!!.ivClose.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view: View) {
                onBackPressed()
            }
        })
        binding!!.tvPrevious.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view: View) {
                if (pageOffset != 1) {
                    pageOffset--
                    getAllCases("")
                }
            }
        })
        binding!!.tvNext.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view: View) {
                if (totalPage != pageOffset) {
                    pageOffset++
                    getAllCases("")
                }
            }
        })
        binding!!.filterIcon.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view: View) {
                val builder = AlertDialog.Builder(this@TruckBookListingActivity)
                val inflater = (this@TruckBookListingActivity as Activity).layoutInflater
                val dialogView = inflater.inflate(R.layout.fiter_diloag, null)
                val notes = dialogView.findViewById<View>(R.id.notes) as EditText
                val submit = dialogView.findViewById<View>(R.id.btn_submit) as Button
                val cancel_btn = dialogView.findViewById<View>(R.id.cancel_btn) as ImageView
                builder.setView(dialogView)
                builder.setCancelable(false)
                val alertDialog = builder.create()
                alertDialog.show()
                cancel_btn.setOnClickListener(object : View.OnClickListener {
                    override fun onClick(v: View) {
                        alertDialog.dismiss()
                    }
                })
                submit.setOnClickListener(object : View.OnClickListener {
                    override fun onClick(v: View) {
                        if (notes.text.toString()
                                .trim { it <= ' ' } != null && !notes.text.toString()
                                .trim { it <= ' ' }
                                .isEmpty()
                        ) {
                            alertDialog.dismiss()
                            pageOffset = 1
                            getAllCases(notes.text.toString().trim { it <= ' ' })
                            //     ClosedPricing(alertDialog, AllCases.get(postion).getCaseId(), notes.getText().toString().trim());
                        } else {
                            Toast.makeText(
                                applicationContext,
                                "Please Fill Text",
                                Toast.LENGTH_LONG
                            ).show()
                        }
                    }
                })
                //  setDateTimeField();
            }
        })
    }

    override fun onBackPressed() {
        super.onBackPressed()
    }

    private fun setAdapter() {
        binding!!.rvDefaultersStatus.addItemDecoration(
            DividerItemDecoration(
                this@TruckBookListingActivity,
                LinearLayoutManager.HORIZONTAL
            )
        )
        binding!!.rvDefaultersStatus.setHasFixedSize(true)
        binding!!.rvDefaultersStatus.isNestedScrollingEnabled = false
        val horizontalLayoutManager =
            LinearLayoutManager(this@TruckBookListingActivity, LinearLayoutManager.VERTICAL, false)
        binding!!.rvDefaultersStatus.layoutManager = horizontalLayoutManager
        truckBookAdapter = TruckBookAdapter(AllCases, this@TruckBookListingActivity, activity)
        binding!!.rvDefaultersStatus.adapter = truckBookAdapter
        //        pagination(horizontalLayoutManager);
    }

    private fun getAllCases(search: String) {
        showDialog()
        truckViewModel.getTruckBookList("10",pageOffset,"IN",search)
        truckViewModel.response.observe(this){
            when(it){

                is NetworkResult.Error -> {
                    hideDialog()
                }
                is NetworkResult.Loading -> {
                    showDialog()
                }
                is NetworkResult.Success ->{
                    binding!!.swipeRefresherStock.isRefreshing = false
                    AllCases!!.clear()
                    if (it.data!!.truckBookCollection == null) {
                        binding!!.txtemptyMsg.visibility = View.VISIBLE
                        binding!!.rvDefaultersStatus.visibility = View.GONE
                        binding!!.pageNextPrivious.visibility = View.GONE
                    } else {
                        // AllCases=body.getTruckBookCollection().getData();
                        AllCases!!.clear()
                        totalPage = it.data.truckBookCollection.lastPage
                        AllCases!!.addAll(it.data.truckBookCollection.data)
                        truckBookAdapter!!.notifyDataSetChanged()
                        //                    binding.rvDefaultersStatus.setAdapter(new TruckBookAdapter(body.getTruckBookCollection(), TruckBookListingActivity.this));
                    }
                    hideDialog()
                }
            }

        }

    }

    fun ViewData(position: Int) {
        val displayRectangle = Rect()
        val window = this.window
        window.decorView.getWindowVisibleDisplayFrame(displayRectangle)
        val builder = AlertDialog.Builder(this@TruckBookListingActivity, R.style.CustomAlertDialog)
        val inflater = this@TruckBookListingActivity.layoutInflater
        val dialogView = inflater.inflate(R.layout.lead_view, null)
        dialogView.minimumWidth = (displayRectangle.width() * 1f).toInt()
        dialogView.minimumHeight = (displayRectangle.height() * 1f).toInt()
        builder.setView(dialogView)
        val alertDialog = builder.create()
        val cancel_btn = dialogView.findViewById<ImageView>(R.id.cancel_btn)
        val lead_id = dialogView.findViewById<TextView>(R.id.lead_id)
        val genrated_by = dialogView.findViewById<TextView>(R.id.genrated_by)
        val customer_name = dialogView.findViewById<TextView>(R.id.customer_name)
        val phone_no = dialogView.findViewById<TextView>(R.id.phone_no)
        val location_title = dialogView.findViewById<TextView>(R.id.location_title)
        val commodity_name = dialogView.findViewById<TextView>(R.id.commodity_name)
        val est_quantity_nmae = dialogView.findViewById<TextView>(R.id.est_quantity_nmae)
        val terminal_name = dialogView.findViewById<TextView>(R.id.terminal_name)
        val purpose_name = dialogView.findViewById<TextView>(R.id.purpose_name)
        val commitemate_date = dialogView.findViewById<TextView>(R.id.commitemate_date)
        val total_weights = dialogView.findViewById<TextView>(R.id.vehicle_no)
        val total_bags = dialogView.findViewById<TextView>(R.id.in_out)
        val create_date = dialogView.findViewById<TextView>(R.id.create_date)
        val case_id = dialogView.findViewById<TextView>(R.id.a1)
        val username = dialogView.findViewById<TextView>(R.id.a2)
        val vehicle_no = dialogView.findViewById<TextView>(R.id.a4)
        val processing_fees = dialogView.findViewById<TextView>(R.id.a5)
        val inteerset_rate = dialogView.findViewById<TextView>(R.id.a6)
        val transport_rate = dialogView.findViewById<TextView>(R.id.a7)
        val loan = dialogView.findViewById<TextView>(R.id.a8)
        val price = dialogView.findViewById<TextView>(R.id.a9)
        val rent = dialogView.findViewById<TextView>(R.id.a10)
        val labour_rent = dialogView.findViewById<TextView>(R.id.a11)
        val total_weight = dialogView.findViewById<TextView>(R.id.a12)
        val notes = dialogView.findViewById<TextView>(R.id.a13)
        val bags = dialogView.findViewById<TextView>(R.id.a14)
        val price_extra = dialogView.findViewById<LinearLayout>(R.id.price_extra)
        price_extra.visibility = View.VISIBLE
        val case_extra = dialogView.findViewById<LinearLayout>(R.id.case_extra)
        case_extra.visibility = View.VISIBLE
        val truck_extra = dialogView.findViewById<LinearLayout>(R.id.truck_extra)
        truck_extra.visibility = View.VISIBLE
        val converted_by = dialogView.findViewById<TextView>(R.id.converted_by)
        val gate_pass = dialogView.findViewById<TextView>(R.id.gate_pass)
        val user = dialogView.findViewById<TextView>(R.id.user)
        val coldwin = dialogView.findViewById<TextView>(R.id.coldwin)
        val purchase_details = dialogView.findViewById<TextView>(R.id.purchase_details)
        val loan_details = dialogView.findViewById<TextView>(R.id.loan_details)
        val selas_details = dialogView.findViewById<TextView>(R.id.selas_details)
        val gate_passs = dialogView.findViewById<TextView>(R.id.gate_passs)
        val total_trans_cost = dialogView.findViewById<TextView>(R.id.total_trans_cost)
        val advance_patyment = dialogView.findViewById<TextView>(R.id.advance_patyment)
        val start_date_time = dialogView.findViewById<TextView>(R.id.start_date_time)
        val end_date_time = dialogView.findViewById<TextView>(R.id.end_date_time)
        val settleememt_amount = dialogView.findViewById<TextView>(R.id.settleememt_amount)
        gate_passs.text = "" + (if ((AllCases!!.get(position)!!
                .gatePass) != null
        ) AllCases!!.get(position)!!.gatePass else "N/A")
        total_trans_cost.text = "" + (if ((AllCases!!.get(position)!!
                .totalTransportCost) != null
        ) AllCases!!.get(position)!!.totalTransportCost else "N/A")
        advance_patyment.text = "" + (if ((AllCases!!.get(position)!!
                .advancePayment) != null
        ) AllCases!!.get(position)!!.advancePayment else "N/A")
        start_date_time.text = "" + (if ((AllCases!!.get(position)!!
                .startDateTime) != null
        ) AllCases!!.get(position)!!.startDateTime else "N/A")
        end_date_time.text = "" + (if ((AllCases!!.get(position)!!
                .endDateTime) != null
        ) AllCases!!.get(position)!!.endDateTime else "N/A")
        settleememt_amount.text = "" + (if ((AllCases!!.get(position)!!
                .finalSettlementAmount) != null
        ) AllCases!!.get(position)!!.finalSettlementAmount else "N/A")
        total_bags.text = "" + (if ((AllCases!!.get(position)!!
                .noOfBags) != null
        ) AllCases!!.get(position)!!.noOfBags else "N/A")
        total_weights.text = "" + (if ((AllCases!!.get(position)!!
                .notes) != null
        ) AllCases!!.get(position)!!.notes else "N/A")
        converted_by.text = "" + (if ((AllCases!!.get(position)!!
                .totalWeight) != null
        ) AllCases!!.get(position)!!.totalWeight else "N/A")
        gate_pass.text = "Gaatepass/CDF Name : " + (if ((AllCases!!.get(position)!!
                .gatePassCdfUserName) != null
        ) AllCases!!.get(position)!!.gatePassCdfUserName else "N/A")
        coldwin.text = "ColdWin Name: " + (if ((AllCases!!.get(position)!!
                .coldwinName) != null
        ) AllCases!!.get(position)!!.coldwinName else "N/A")
        user.text = "User : " + (if ((AllCases!!.get(position)!!
                .fpoUserId) != null
        ) AllCases!!.get(position)!!.fpoUserId else "N/A")
        purchase_details.text = "purchase Details : " + (if ((AllCases!!.get(position)!!
                .purchaseName) != null
        ) AllCases!!.get(position)!!.purchaseName else "N/A")
        loan_details.text = "Loan Details : " + (if ((AllCases!!.get(position)!!
                .loanName) != null
        ) AllCases!!.get(position)!!.loanName else "N/A")
        selas_details.text = "Sale Details : " + (if ((AllCases!!.get(position)!!
                .saleName) != null
        ) AllCases!!.get(position)!!.saleName else "N/A")
        total_weight.text = resources.getString(R.string.total_weight)
        bags.text = resources.getString(R.string.bags)
        case_id.text = resources.getString(R.string.case_idd)
        notes.text = resources.getString(R.string.notes)
        rent.text = resources.getString(R.string.turnaround_time)
        labour_rent.text = resources.getString(R.string.commodity)
        loan.text = resources.getString(R.string.min_weight)
        transport_rate.text = resources.getString(R.string.transport_rate)
        price.text = resources.getString(R.string.max_weight)
        vehicle_no.text = resources.getString(R.string.vehicle_noa)
        processing_fees.text = resources.getString(R.string.driver_name)
        inteerset_rate.text = resources.getString(R.string.driver_phone_no)
        username.text = resources.getString(R.string.transporter)
        lead_id.text = "" + AllCases!!.get(position)!!.caseId
        genrated_by.text = "" + (if ((AllCases!!.get(position)!!
                .transporter) != null
        ) AllCases!!.get(position)!!.transporter else "N/A")
        customer_name.text = "" + AllCases!!.get(position)!!.custFname
        location_title.text = "" + (if ((AllCases!!.get(position)!!
                .driverName) != null
        ) AllCases!!.get(position)!!.driverName else "N/A")
        phone_no.text = "" + (if ((AllCases!!.get(position)!!
                .vehicle) != null
        ) AllCases!!.get(position)!!.vehicle else "N/A")
        commodity_name.text = "" + (if ((AllCases!!.get(position)!!
                .driverPhone) != null
        ) AllCases!!.get(position)!!.driverPhone else "N/A")
        est_quantity_nmae.text = "" + (if ((AllCases!!.get(position)!!
                .ratePerKm) != null
        ) AllCases!!.get(position)!!.ratePerKm else "N/A")
        terminal_name.text = "" + (if ((AllCases!!.get(position)!!
                .minWeight) != null
        ) AllCases!!.get(position)!!.minWeight else "N/A")
        purpose_name.text = "" + (if ((AllCases!!.get(position)!!
                .maxWeight) != null
        ) AllCases!!.get(position)!!.maxWeight else "N/A")
        commitemate_date.text = "" + (if ((AllCases!!.get(position)!!
                .turnaroundTime) != null
        ) AllCases!!.get(position)!!.turnaroundTime else "N/A")
        create_date.text = "" + (if ((AllCases!!.get(position)!!
                .commodityId) != null
        ) AllCases!!.get(position)!!.commodityId else "N/A")
        val truck_file = dialogView.findViewById<View>(R.id.Bilty_Image) as ImageView
        val biltyLL = dialogView.findViewById<View>(R.id.biltyLL) as LinearLayout
        biltyLL.visibility = View.VISIBLE
        if (AllCases!![position]!!.file == null || AllCases!![position]!!.file.isEmpty()) {
            truck_file.visibility = View.GONE
        }
        TruckImage = Constants.Truck_bilty_photo + AllCases!![position]!!
            .file
        truck_file.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view: View) {
                PhotoFullPopupWindow(
                    this@TruckBookListingActivity,
                    R.layout.popup_photo_full,
                    view,
                    TruckImage,
                    null
                )
            }
        })
        cancel_btn.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View) {
                alertDialog.dismiss()
            }
        })
        alertDialog.show()
    }

    fun checkVeehicleNo(postion: Int) {
        val bundle = Bundle()
        bundle.putString("user_name", AllCases!![postion]!!.custFname)
        bundle.putString("case_id", AllCases!![postion]!!.caseId)
        bundle.putString("vehicle_no", AllCases!![postion]!!.vehicleNo)
        startActivity(TruckUploadDetailsClass::class.java, bundle)
        /* apiService.cheeckvehiclePricicng(AllCases.get(postion).getCaseId()).enqueue(new NetworkCallback<VehcilePricingCheeck>(getActivity()) {
            @Override
            protected void onSuccess(VehcilePricingCheeck body) {
                if (body.getVichel_status().equalsIgnoreCase("1")) {
                    Toast.makeText(getApplicationContext(), body.getMessage(), Toast.LENGTH_LONG).show();
                } else {

                }
            }
        });*/
    }
}
